caluma/caluma_logging/middleware.py

```diff
@@ -0,0 +1,58 @@
+import json
+
+from graphql.error.syntax_error import GraphQLSyntaxError
+from graphql.language import parser, visitor
+
+from caluma.caluma_logging.models import AccessLog
+
+
+class AccessLogMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        response = self.get_response(request)
+        # user is available only after the request was processed
+        try:
+            body = json.loads(request.body.decode("utf-8"))
+        except json.decoder.JSONDecodeError:
+            return response
+
+        vis = AccessLogVisitor()
+        try:
+            doc = parser.parse(body["query"])
```

Comment:
Aaah I like this one! Except for the fact that now the whole request parsing will happen twice. But we can leave that optimisation for another time, I don't think it's worth it to dig too deep just now